rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -142,7 +142,10 @@ rcl_logging_ret_t rcl_logging_external_initialize(
       RCUTILS_SET_ERROR_MSG("Failed to create log file name string");
       return RCL_LOGGING_RET_ERROR;
     }
-    g_root_logger = spdlog::basic_logger_mt("root", name_buffer);
+
+    auto sink = std::make_shared<spdlog::sinks::basic_file_sink_mt>(name_buffer, false);
```

Comment:
I think we can clarify the ownership semantics here by making this a `std::unique_ptr`.

Reply:
Are you referring to the `sink` or the `g_root_logger`?
About the first, the function takes a `shared_ptr`. IMO, creating and moving any of both smart pointers is equally clear (and should produce the same code), but I changed to `unique_ptr`.
About the second, I moved to an `unique_ptr`.

Reply:
Looks good, thanks.